### ISSUE TITLE
[Proposal] Add alias StateValues for Values

### ISSUE THREAD

**Issue by itscaro** [roboll/helmfile#1743]
IMO it would be nice to have another variable name for state values such as `.StateValues` as alias of `.Values`. In my usage, I have a lot of raw content which content `.Values` which is helm variable, having both helmfile `.Values` and helm `.Values` in the same file is sometimes confusing.

**Comment by antaloala** [comment]
@itscaro what is your usecase?
I am not able to see how helmfile state values and helm values can land in the same templated yaml manifest...

**Comment by itscaro** [comment]
I use gotmpl and inside helmfile value are used, for example, to template ingress dns. In the same file I have a list of K8S manifest like volumes which will be rendered by helm later. These configmap cannot be rendered by helmfile because helmfile `.Values` does not contain the values which it is generating for helm.

```
# .Values is helmfile's one
image:
  tag: {{.Values.app_1.version}}

ingress:
  enabled: true
  hostname: {{.Values.stack.prefix}}-{{.Environment.Name}}.{{.Values.dnsBase}}

# .Values is helm's one
# Raw content: helm will do the render
{{`
extraVolumeMounts: |
  {{- if .Values.metrics.enabled }}
  - name: metrics
    mountPath: /metrics
  {{- end }}
`}}

```

Other solutions to avoid having `.Values` of two nature:
- declare locally a new variable `{{ $StateValues := .Values }}`, this is ok for me but having a native alias is way better ;-)
- separate into multiple files, but this solution generates too many files to handle and not fits best for me.

**Comment by mumoshu** [comment]
@itscaro Hey! I looked your example but still curious how it could work.

Even if you named StateValues as `.StateValues`, any reference to `{{ .Values }}` would break your deployment as either (1)your helmfile template would fail due to missing values(because helmfile looks for state values rather than helm chart values) or (2)helm doesn't seen any `{{ .Values | whatever }}` expressions as helmfile has already rendered and replaced all of them.

Maybe I'm missing something? Could you share an example project that doesn't work before but starts working after #1744?

**Comment by itscaro** [comment]
@mumoshu In the above example you can see there is usage of two .Values which do not mean the same thing. The .Values out of raw content are values coming from environment context of helmfile, and in raw part the .Values is helm's one which is values generated from helmfile.

**Comment by mumoshu** [comment]
@itscaro Ah sorry I somehow missed the beginning and the end of the template comment in the last half of your example. All good!
